scripts/traverse.js

```diff
@@ -2,89 +2,143 @@
 const bcd = require('..');
 
 const { argv } = require('yargs').command(
-  '$0 <browser> [folder] [value]',
-  'Test for specified values in any specified browser',
+  '$0 [folder...]',
+  'Print feature names in the folder (and optionally filter features to specific browser or version values)',
   yargs => {
     yargs
-      .positional('browser', {
-        describe: 'The browser to test for',
-        type: 'string',
-      })
       .positional('folder', {
-        describe: 'The folder(s) to test (set to "all" for all folders)',
+        describe: 'The folder(s) to traverse',
+        type: 'array',
+        default: Object.keys(bcd).filter(k => k !== 'browsers'),
+      })
+      .option('browser', {
+        alias: 'b',
+        describe: 'Filter by a browser. May repeat.',
         type: 'array',
-        default: 'all',
+        nargs: 1,
+        default: Object.keys(bcd.browsers),
       })
-      .positional('value', {
-        describe: 'The value(s) to test against',
+      .option('filter', {
+        alias: 'f',
+        describe: 'Filter by version value. May repeat.',
         type: 'array',
-        default: ['null', 'true'],
+        nargs: 1,
+        default: [],
+      })
+      .option('non-real', {
+        describe:
+          'Filter to features with non-real values. Alias for "-f true -f null"',
+        type: 'boolean',
+        nargs: 0,
       })
       .option('depth', {
         alias: 'd',
         describe:
           'Depth of features to traverse (ex. "2" will capture "api.CSSStyleSheet.insertRule" but not "api.CSSStyleSheet.insertRule.optional_index")',
         type: 'number',
+        nargs: 1,
         default: 100,
-      });
+      })
+      .example(
+        'npm run traverse -- --browser=safari --non-real',
+        'Find all features containing non-real Safari entries',
+      )
+      .example(
+        'npm run traverse -- -b webview_android -f true',
+        'Find all features marked as true for WebVeiw',
+      )
+      .example(
+        'npm run traverse -- -b firefox -f 10',
+        'Find all features marked as supported since Firefox 10',
+      );
   },
 );
 
-function traverseFeatures(obj, depth, identifier) {
+function traverseFeatures(obj, browsers, values, depth, identifier) {
```

Comment:
What do you think about exposing `traverseFeatures` for other scripts to use when not invoking this script directly? That would also make it easy to write tests for this.